fire/core_test.py

```diff
@@ -194,6 +194,28 @@ def testClassMethod(self):
         7,
     )
 
+  def testCustomSerialize(self):
+    def serialize(x):
+      if isinstance(x, list):
+        return ', '.join(str(xi) for xi in x)
+      if isinstance(x, dict):
+        return ', '.join('{}={!r}'.format(k, v) for k, v in x.items())
+      if x == 'special':
+        return ['SURPRISE!!', "I'm a list!"]
```

Comment:
The name "serialize" implies that the output of serialize is either bytes or a string.
But that's not what we're requiring.
I wonder if there's a better name (maybe that's "format" after all.)
I'll think on this, but curious for your thoughts.

Reply:
Maybe "process" or "finalize"...
I'll keep thinking.

Reply:
Forgive my unpolished thinking aloud:

Maybe "serialize" is fine, and the logic is: "if you don't finish serializing it, we'll continue serializing it for you using the default serializer."
And then "display" will only ever be given bytes or a string, rather than being directly given the output of serialize.
One challenge with this is that it disallows different serialization for different display modes.
We could (in a subsequent PR) allow serialize to accept the "kind" parameter too to reenable this capability.

Reply:
side note: one thing that would be nice is if we could have some sort of override via the CLI where the user could do `--format json` or `--format csv` or something like that

Reply:
Some notes about the naming - I like `serialize` and `format` because they both imply a string result. I think there might be a problem with `process` or `finalize` cuz they could imply that you have the ability to modify the result after each iteration (after each getattr / getitem / call).

I can't think of any other words better than those two tho.

> Maybe "serialize" is fine, and the logic is: "if you don't finish serializing it, we'll continue serializing it for you using the default serializer."

I think this is fine because even if the user doesn't realize that they can't return a string from it, it's not going to break anything, it might just make a little more work for them.